src/Browser.php

```diff
@@ -335,4 +337,32 @@ public function withOptions(array $options)
 
         return $browser;
     }
+
+    public function withTimeout($timeout)
+    {
+        return $this->withOptions(array(
+            'timeout' => $timeout,
+        ));
+    }
+
+    public function withFollowRedirects($followRedirects)
+    {
+        return $this->withOptions(array(
+            'followRedirects' => $followRedirects,
+        ));
+    }
+
+    public function withMaxRedirects($maxRedirects)
+    {
+        return $this->withOptions(array(
+            'maxRedirects' => $maxRedirects,
+        ));
+    }
+
+    public function withObeySuccessCode($obeySuccessCode)
```

Comment:
Docblock please? This would probably accept a `bool`, does `withObeySuccessCode(false)` make sense semantically?

Reply:
To be fair, does that feature make sense it all? Should this HTTP client care about the response that much that it turns anything about 400? In my opinion, it's like cache, any result from the cache engine is valid, and only communication errors between the cache and the adapter should yield exceptions.

Reply:
I think you're raising a valid point that's worth discussing, but I see this is still supported in other projects like https://www.php.net/manual/de/context.http.php#context.http.ignore-errors and http://docs.guzzlephp.org/en/stable/request-options.html#http-errors.

I would suggest adding this as part of this PR so we can deprecate the `withOptions()` method and retain all functionality. If you feel it's worth it, we can still discuss this in a separate ticket for a possible v3 :+1:

Reply:
Wasn't intending on removing it in this PR, already added the docblock for it, but I do feel it's worth removing from v3.

Reply:
Keeping it makes sense, since it allows cleaner code when using the library and the handling code for all HTTP response code is the same. This would allow to handle only "real errors" in a promise rejection handler, while being able to handle completed HTTP requests (regardless of their status code) in one handler, without boilerplate code.